docs/README-CRD.md

```diff
@@ -0,0 +1,95 @@
+# How to write CRD that leverage the capabilities of the dashboard
+
+Although the dashboard is object/agnostic, i.e. it can automatically show any object that is available in Kubernetes, the usability is greatly impacted by the way CRDs are created.
+
+This document presents some suggestions about how to write CRDs in a way that facilitate the operations of the dashboard.
+
+## Objects and fields description
+The dashboard leverages the presence of textual descriptions of a given field/object to show a tooltip that help understanding what that object is about.
+
+**Example of a traditional CRD**
+```
+...
+resources:
+  type: object
+  properties:
+    cpu:
+      type: integer
+      maximum: 8
+      minimum: 1
+    memory:
+      type: string
+    reservedCpuPercentage:
+      type: integer
+      maximum: 100
+      minimum: 1
+...
+```
+
+![CRD-without-field-description](images/CRD-without-field-description.png)
+
+**Example of an enriched CRD**
+
+```
+...
+resources:
+  description: The resources to assign to the resource
+  type: object
+  properties:
+    cpu:
+      description: Number of cores to assign to the resource (from 1 to 8)
+      type: integer
+      maximum: 8
+      minimum: 1
+    memory:
+      description: How much RAM (in units of M or G) to assign to the resource
+      type: string
```

Comment:
I turned it into "string" because it was easier to read that way, as the key point here is to implement the "description" field.

Reply:
Gotcha. I would maybe add `pattern: ...` to make it clear that you are omitting it since not relevant for the example. Otherwise it may look strange at first sight